pkg/api/types.go

```diff
@@ -158,13 +158,101 @@ type VolumeSource struct {
 	HostDir *HostDir `json:"hostDir"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDir `json:"emptyDir"`
+	// GitRepo represents a git repository at a particular revision.
+	GitRepo *GitRepo `json:"gitRepo"`
+
 	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
 	GCEPersistentDisk *GCEPersistentDisk `json:"persistentDisk"`
-	// GitRepo represents a git repository at a particular revision.
-	GitRepo *GitRepo `json:"gitRepo"`
+
+	// IMPORTANT!  GCEPersistentDisk will be refactored into PersistentDisk.
+	// For this proposal, it is left in for compilation purposes.
+
+	// DiskSpec represents a request for persistent storage that is attached to
+	// kubelet's host machine and then exposed to the pod.
+	// The specific implementation of storage is obtained through the cloud provider.
+	PersistentDisk *PersistentDisk `json:diskSpec`
+}
+
+type PersistentDisk struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+	Spec       DiskSpec   `json:"spec,omitempty"`
+	Status     DiskStatus `json:"status,omitempty"`
 }
 
+type DiskSpec struct {
+
+	// unique name of the PD resource.
+	// used to identify the disk in GCE or AWS
+	PDName string `json: "pdName"`
```

Comment:
Yes, the structs are woefully out of date (not to mention formatting).  They are leftover from the previous design and need to be reworked entirely for the new design laid out in persistentdisks.md.  Feedback on that doc is greatly appreciated!  :)